storage.go

```diff
@@ -349,7 +355,11 @@ func (s *Storage) reach(ctx context.Context, path string, opt pairStorageReach)
 }
 
 func (s *Storage) read(ctx context.Context, path string, w io.Writer, opt pairStorageRead) (n int64, err error) {
-	input := &service.GetObjectInput{}
+	input := &service.GetObjectInput{
+		XQSEncryptionCustomerAlgorithm: &opt.SseCustomerAlgorithm,
```

Comment:
We need to check value before use them, please check `HasSseCustomerAlgorithm` first.